syntax = "proto3";
package prescan.control;

import "bridge_msgs/basic_msgs/header.proto";
import "bridge_msgs/chassis_msgs/chassis.proto";


// next id : 27
message ControlCommand {
  prescan.common.Header header = 1;
  // target throttle in  [0, 1]
  double throttle = 3;

  // target brake in [0, 1]
  double brake = 4;

  // target steering angle in [-1, 1]
  double steering = 7;

  // target speed, in m/s
  double tar_speed = 9;

  // target acceleration in m`s^-2
  double tar_acceleration = 10;

  // target GearPosition
  prescan.chassis.Chassis.GearPosition gear_location = 20;
}
